component/discovery/kubernetes/configtypes.go

```diff
@@ -0,0 +1,151 @@
+package kubernetes
+
+import (
+	"net/url"
+
+	"github.com/prometheus/common/config"
+)
+
+// TODO (cpeterson) move HTTPClientConfig and subtypes to dedicated package. make sure all custom serializers work
+type HTTPClientConfig struct {
```

Comment:
I also had to implement the HTTPClientConfig from Prometheus [here](https://github.com/grafana/agent/blob/e24c6fab53130b51b8387562322cef1f7d3f97f1/component/metrics/scrape/types.go#L65-L69); we should definitely be reusing them in a shared package.

Reply:
Also, might be a good idea to call the HTTPClientConfig's [Validate method](https://github.com/prometheus/common/blob/49b36038aead362ce78f09337dc5872000bd048a/config/http_config.go#L278-L280) to check that the user didn't define conflicting auth mechanisms.

Reply:
Yes, now is definitely the right time to make a shared package. Any suggestions for the right import path?

Reply:
Could be under `component/common` but I'm not sure about the package name. `promhttpclientconfig` sounds unwieldy; `promhttp` maybe?